hoomd/data/typeparam.py

```diff
@@ -52,14 +55,15 @@ def to_dict(self):
     def keys(self):
         yield from self.param_dict.keys()
 
-    @property
-    def state(self):
-        state = self.to_dict()
-        if self.param_dict._len_keys > 1:
-            state = {str(key): value for key, value in state.items()}
-        state['__default__'] = self.default
+    def __getstate__(self):
+        state = {'name': self.name,
+                 'type_kind': self.type_kind,
+                 'param_dict': self.param_dict
+                 }
+        if isinstance(self.param_dict, AttachedTypeParameterDict):
+            state['param_dict'] = self.param_dict.to_dettached()
```

Comment:
Typo? (Missing test?)
```suggestion
            state['param_dict'] = self.param_dict.to_detached()
```

Reply:
Typo, a consistent one.